File: app/src/main/java/life/plank/juna/zone/view/adapter/multiview/binder/ScrubberBinder.java
package life.plank.juna.zone.view.adapter.multiview.binder;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ahamed.multiviewadapter.ItemBinder;
import com.ahamed.multiviewadapter.ItemViewHolder;
import com.github.mikephil.charting.charts.LineChart;

import java.lang.ref.WeakReference;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import life.plank.juna.zone.R;
import life.plank.juna.zone.data.model.binder.ScrubberBindingModel;
import life.plank.juna.zone.util.DataUtil.ScrubberLoader;
import life.plank.juna.zone.view.adapter.multiview.BoardInfoAdapter;

public class ScrubberBinder extends ItemBinder<ScrubberBindingModel, ScrubberBinder.ScrubberViewHolder> {

    private BoardInfoAdapter.BoardInfoAdapterListener listener;

    public ScrubberBinder(BoardInfoAdapter.BoardInfoAdapterListener listener) {
        this.listener = listener;
    }

    @Override
    public ScrubberViewHolder create(LayoutInflater inflater, ViewGroup parent) {
        return new ScrubberViewHolder(inflater.inflate(R.layout.item_scrubber, parent, false), this);
    }

    @Override
    public void bind(ScrubberViewHolder holder, ScrubberBindingModel item) {
        ScrubberLoader.prepare(holder.scrubber, false);
    }

    @Override
    public boolean canBindData(Object item) {
        return item instanceof ScrubberBindingModel;
    }

    static class ScrubberViewHolder extends ItemViewHolder<ScrubberBindingModel> {

        @BindView(R.id.scrubber)
        LineChart scrubber;

        private final WeakReference<ScrubberBinder> ref;

        ScrubberViewHolder(View itemView, ScrubberBinder scrubberBinder) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            ref = new WeakReference<>(scrubberBinder);
        }

        @OnClick(R.id.scrubber)
        public void onScrubberClick() {
            ref.get().listener.onScrubberClick(scrubber);
        }
    }
}

File: app/src/main/java/life/plank/juna/zone/view/adapter/board/match/binder/ScrubberBinder.kt
package life.plank.juna.zone.view.adapter.board.match.binder

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.ahamed.multiviewadapter.ItemBinder
import com.ahamed.multiviewadapter.ItemViewHolder

import kotlinx.android.synthetic.main.item_scrubber.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.data.model.binder.ScrubberBindingModel
import life.plank.juna.zone.util.DataUtil.ScrubberLoader
import life.plank.juna.zone.util.facilis.onDebouncingClick
import life.plank.juna.zone.view.adapter.board.match.BoardMediaAdapter

class ScrubberBinder(private val listener: BoardMediaAdapter.BoardMediaAdapterListener) : ItemBinder<ScrubberBindingModel, ScrubberBinder.ScrubberViewHolder>() {

    override fun create(inflater: LayoutInflater, parent: ViewGroup): ScrubberViewHolder = ScrubberViewHolder(inflater.inflate(R.layout.item_scrubber, parent, false))

    override fun bind(holder: ScrubberViewHolder, item: ScrubberBindingModel) {
        holder.itemView.run {
            ScrubberLoader.prepare(scrubber, false)
            scrubber.onDebouncingClick { listener.onScrubberClick(scrubber) }
        }
    }

    override fun canBindData(item: Any): Boolean = item is ScrubberBindingModel

    class ScrubberViewHolder(itemView: View) : ItemViewHolder<ScrubberBindingModel>(itemView)
}
